sc-memory/sc-memory/src/scs/scs_parser.cpp

```diff
@@ -480,22 +480,24 @@ void Parser::ProcessTriple(
     if (source.IsElementType() && (connector.GetType() == ScType::ConstPermPosArc))
     {
       std::string const & sourceIdtf = source.GetIdtf();
-      ScType const newType = target.m_type | scs::TypeResolver::GetKeynodeType(sourceIdtf);
+      ScType const sourceType = scs::TypeResolver::GetElementType(sourceIdtf);
+      ScType const targetType = target.m_type;
+      ScType const newTargetType = targetType | sourceType;
 
-      if (target.IsElementType() && !newType.CanExtendTo(ScType::ConstNodeClass))
+      if (target.IsElementType() && !newTargetType.CanExtendTo(ScType::ConstNodeClass))
         SC_THROW_EXCEPTION(
             utils::ExceptionParseError,
-            "Can't extend type `" << std::string(target.m_type) << "` to type `" << std::string(newType)
-                                  << "` for element `" << target.GetIdtf() << "`, because element `" << target.GetIdtf()
-                                  << "` is element denoting type of sc-elements.");
-      else if (target.m_type.CanExtendTo(newType))
-        target.m_type = newType;
+            "Can't extend type `" << std::string(targetType) << "` to type `" << std::string(sourceType)
+                                  << "` for specified sc-element, because sc-element `" << target.GetIdtf()
+                                  << "` is sc-element denoting type of sc-elements.");
+      else if (ScType(targetType.BitAnd(~(ScType::Const | ScType::Var))).CanExtendTo(sourceType))
+        target.m_type = newTargetType;
       else
         SC_THROW_EXCEPTION(
```

Comment:
I've added membership arc to sc_membership_arc and got an error
![image](https://github.com/user-attachments/assets/f80b1354-99ac-4ee3-b10c-640e89580049)
![image](https://github.com/user-attachments/assets/68cbc65d-9b76-4949-8d80-53754eef3014)

Reply:
please add tests

Reply:
TEST_F(SCsHelperTest, GenerateBySCs_MembershipArcBelongsToCommonArcsType)

Reply:
See the latest commit